plugins/module_utils/network/sonic/config/route_maps/route_maps.py

```diff
@@ -0,0 +1,2094 @@
+                           'openconfig-routing-policy-ext:match-src-network-instance'),
+            'tag': (match_delete_req_base +
+                    'match-tag-set/config/openconfig-routing-policy-ext:tag-value')
+        }
+
+        # Remove all appropriate "match" configuration for this route map if any of the
+        # following criteria are met:  (See the note below regarding what configuration
+        # is "appropriate"for deletion.)
+        #
+        # 1) Any top level attribute is specified with a value different from its current
+        #    configured value.
+        # 2) Any top level attribute is specified that is not currently configured.
+        # 3) The set of top level attributes specified does not include all currently
+        #    configured attributes (regardless of whether the specified values for
+        #    these attributes are the same as the ones courrently configured).
+        # (Note: Although the IPv6 attribute is defined as a nested dictionary
+        # to allow for future expansion, it is handled here as a top level
+        # attrbute because it currently has only one member.)
+        #
+        # When deletion has been triggered, an attribute is deleted only if it is
+        # not present at all in the requested configuration. (If it is present in
+        # the requested configuration, the "merge" phase of the "replaced" state
+        # operation will modify it as needed, so it doesn not need to be explicitly
+        # deleted during the "deletion" phase.)
+        #
+        cmd_top_level_key_set = set(cmd_match_keys).intersection(set(match_top_level_keys))
+        cfg_top_level_key_set = set(cfg_match_keys).intersection(set(match_top_level_keys))
+        symmetric_diff_set = cmd_top_level_key_set.symmetric_difference(cfg_top_level_key_set)
+        intersection_diff_set = cmd_top_level_key_set.intersection(cfg_top_level_key_set)
+        cmd_delete_dict = {}
+        if (cmd_top_level_key_set and symmetric_diff_set or
+                (any(keyname for keyname in intersection_diff_set if
+                     cmd_match_top[keyname] != cfg_match_top[keyname]))):
+            for cfg_match_key in cfg_top_level_key_set.difference(cmd_top_level_key_set):
+                match_uri = match_uri_attr[cfg_match_key]
+                cmd_delete_dict[cfg_match_key] = cfg_match_top[cfg_match_key]
+                if isinstance(match_uri, dict):
+                    for member_key in match_uri:
+                        if cfg_match_top[cfg_match_key].get(member_key):
+                            request = {'path': match_uri[member_key], 'method': DELETE}
+                            requests.append(request)
+                else:
+                    request = {'path': match_uri, 'method': DELETE}
+                    requests.append(request)
+            command.pop('match')
+            if cmd_delete_dict:
+                command['match'] = cmd_delete_dict
+            return
```

Comment:
If any one of 'match_top_level_keys' is to be deleted, then deletion of valid entries in 'match_multi_level_keys' is skipped. Please ensure other match options are handled, when there is deletion of option in 'match_top_level_keys'.

Reply:
Done, here and also for "set" top level deletion.